docs/bsp.md

```diff
@@ -902,14 +1077,69 @@ trait TestReport {
 }
 ```
 
-The `TestReport` notification will be sent as the test execution of build targets completes.
+This request may trigger a compilation or other tasks on the selected build targets.
 
-This request may trigger a compilation on the selected build targets. The server is free to send any
-number of `build/compileReport`, `build/publishDiagnostics` and `build/logMessage` notifications
-during compilation before completing the response.
+The server may send any number of `build/task*`, `build/publishDiagnostics` and `build/logMessage` notifications
+to communicate about tasks triggered by the request to the client.
+
+#### Test Unit Notifications
+
+The server may inform about individual test units in task notifications that reference the originating task in their `taskId`.
+For example, the server can send a `taskStart`/`taskFinish` for each test suite in a target, and likewise for each test in the suite.
+The server's implementation decides the granularity at which tests are reported. For example, if it only has information 
+about all the tests in a suite at a time, it could report a TestFinish for each test once the suite is done.
```

Comment:
Can we update the scaladoc of `TestStart` and `TestFinish`?

Reply:
What do you mean? :)